int32 x
int32 y
uint8[] covariance_matrix
int32 feature_index
